chat-plugins/uno.js

```diff
@@ -601,98 +740,111 @@ exports.commands = {
 
 		autostart: function (target, room, user) {
 			if (!this.can('minigame', null, room)) return;
-			if (!room.game || room.game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
+			const game = /** @type {UnoGame} */ (room.game);
+			if (!game || game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
 			if (toId(target) === 'off') {
-				if (!room.game.autostartTimer) return this.errorReply("There is no autostart timer running on.");
+				if (!game.autostartTimer) return this.errorReply("There is no autostart timer running on.");
 				this.addModAction(`${user.name} has turned off the UNO autostart timer.`);
-				clearTimeout(room.game.autostartTimer);
+				clearTimeout(game.autostartTimer);
 				return;
 			}
 			const amount = parseInt(target);
 			if (!amount || amount < 30 || amount > 600) return this.errorReply("The amount must be a number between 30 and 600 seconds.");
-			if (room.game.state !== 'signups') return this.errorReply("The game of UNO has already started.");
-			if (room.game.autostartTimer) clearTimeout(room.game.autostartTimer);
-			room.game.autostartTimer = setTimeout(() => {
-				room.game.onStart();
+			if (game.state !== 'signups') return this.errorReply("The game of UNO has already started.");
+			if (game.autostartTimer) clearTimeout(game.autostartTimer);
+			game.autostartTimer = setTimeout(() => {
+				game.onStart();
 			}, amount * 1000);
 			this.addModAction(`${user.name} has set the UNO autostart timer to ${amount} seconds.`);
 		},
 
 		dq: 'disqualify',
 		disqualify: function (target, room, user) {
 			if (!this.can('minigame', null, room)) return;
-			if (!room.game || room.game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
+			const game = /** @type {UnoGame} */ (room.game);
+			if (!game || game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
 
-			let disqualified = room.game.eliminate(toId(target));
+			let disqualified = game.eliminate(toId(target));
 			if (disqualified === false) return this.errorReply(`Unable to disqualify ${target}.`);
 			this.privateModAction(`(${user.name} has disqualified ${disqualified} from the UNO game.)`);
 			this.modlog('UNO DQ', toId(target));
 			room.add(`${target} has been disqualified from the UNO game.`).update();
 		},
 
 		// player/user commands
-		j: 'join',
-		join: function (target, room, user) {
-			if (!room.game || room.game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
+		j: 'unojoin',
+		// TypeScript doesn't like 'join' being defined as a function
+		join: 'unojoin',
```

Comment:
Not anything you can fix but do we know why this is happening? I recall trying to figure this out for mafia with zarel.

Reply:
Yeah I have no clue why this happens, I looked at your Mafia PR while doing this to see if the same issue happened and then did this to mirror your fix. I'll address the other comments in a bit.